### PR TITLE
Rules are correctly created, replaced, and deleted

### PR DESCRIPTION
**Pull request by vanstee**

### CODE REVIEW

**Comment by mpeck** [comment]
Is this something we want to do? I was thinking that bundle rules would have to be updated through bundle installation. Otherwise wouldn't you just overwrite them if you updated a bundle?

File: lib/cog_api/http/rules.ex [hunk 1]
```diff
@@ -17,6 +17,16 @@ defmodule CogApi.HTTP.Rules do
     |> ApiResponse.format(%Rule{})
   end
 
+  def show(rule_id, %Endpoint{}=endpoint) do
+    Base.get(endpoint, "rules/#{rule_id}")
+    |> ApiResponse.format(%Rule{})
+  end
+
+  def update(rule_id, rule_text, %Endpoint{}=endpoint) do
+    Base.patch(endpoint, "rules/#{rule_id}", %{rule: rule_text})
```
Comment (mpeck) [inline lib/cog_api/http/rules.ex]: ok, so Cog handles the whole disable and create bit for you?

Comment (vanstee) [inline lib/cog_api/http/rules.ex]: Yep, makes it nice and atomic without having overlapping or missing rules in the middle of an update.

**Comment by mpeck** [comment]
Looks reasonable to me. One thing to note, and I'm not suggesting you change it here, I've been kinda ignoring the client module and just calling functions directly from the module that I need. I think the client module bit adds some unnecessary obfuscation.

**Comment by mpeck** [comment]
:+1:

**Comment by vanstee** [comment]
Yeah I'm not a fan of the client module either, especially now that `fake` is gone. But I've been trying to keep it up to date, might not be worth it.

**Pull request merged**